### PR TITLE
Add web component version of widget

### PR DESCRIPTION
**Pull request by toddkao**
Add web component version of widget, using https://www.npmjs.com/package/@r2wc/react-to-web-component

image

Add example usage of web component to examples page

### CODE REVIEW

File: examples/nextjs/pages/index.tsx [hunk 1]
```diff
@@ -22,6 +32,25 @@ const Home: NextPage = () => {
...
+            colors={JSON.stringify({
+              primary: '#FF4FFF',
+            })}
+            defaultRoute={JSON.stringify({
+              srcChainID: 'osmosis-1',
+              srcAssetDenom:
+                'ibc/1480b8fd20ad5fcae81ea87584d269547dd4d436843c1d20f15e00eb64743ef4',
+            })}>
+          </swap-widget>
+        </NoSSR>
```
Comment (toddkao) [inline examples/nextjs/pages/index.tsx:52]: nextjs does not play nicely with web-components, so i had to disable it to render without errors

Comment (toddkao) [reply examples/nextjs/pages/index.tsx:52]: The other thing that's annoying is the way these attributes are passed, I think I can add a wrapper component and convert these props to individual strings to make it nicer to use

File: packages/widget/src/index.ts [hunk 2]
```diff
@@ -1,5 +1,6 @@
 import './styles/global.css';
 export { SwapWidgetProvider } from './provider';
 export { SwapWidget, SwapWidgetProps } from './ui';
+import './ui/WebComponent';
```
Comment (toddkao) [inline packages/widget/src/index.ts:4]: just added it here temporarily but I think it's better to have the web-component loader in a separate import

File: examples/nextjs/pages/index.tsx (hunk 1 above)
Comment (o-az) [reply examples/nextjs/pages/index.tsx]: Checkout https://npm.im/@lit/react

[`lit`](https://lit.dev/) is a fantastic web components framework

[`@lit/react`](https://npm.im/@lit/react) addresses directly the annoyance of passing attributes you mention.

Comment (toddkao) [reply examples/nextjs/pages/index.tsx]: > Checkout https://npm.im/@lit/react
> 
> [`lit`](https://lit.dev/) is a fantastic web components framework
> 
> [`@lit/react`](https://npm.im/@lit/react) addresses directly the annoyance of passing attributes you mention.

Hey @o-az 👋 I took a look at [lit](https://lit.dev/) and [@lit/react](https://www.npmjs.com/package/@lit/react) and from what I understand from reading through the docs is basically is for creating web-components from scratch, and @lit/react is for **consuming/using** them within react.


The lib i'm currently using, [@r2wc](https://www.npmjs.com/package/@r2wc/react-to-web-component) helps to actually take an existing react-component and turn it into a web-component, which I don't believe lit actually does

File: packages/widget/rollup.config.js [hunk 3]
```diff
@@ -7,7 +7,7 @@ import packageJson from './package.json';
 export default [
   {
     input: ['./src/index.ts'],
-    external: ['react', 'react-dom'],
```
Comment (codingki) [inline packages/widget/rollup.config.js:10]: I don't think you need to make `@r2wc/react-to-web-component` in external packages

**Review by codingki — APPROVED** [review]

**Pull request merged**